### PR TITLE
build: Cleanup cmake options, fix detray plugin

### PR DESCRIPTION
**Pull request by paulgessinger**
- builtin boost is remove
- extern sources set up to actually check hashes
- detray sets up dfelibs (was removed)
- fix ordering: algebra-plugins needs to go before detray
- traccc doesn't specify cmake minimum version
- reorder options to be more readable
- preset tries to enable everything in the build
- detray and covfie plugins are removed in favor of a single traccc plugin

Blocked by:
- #3511

### CODE REVIEW

**Review by andiwand — DISMISSED** [review]
🧹

**Comment by benjaminhuth** [comment]
I wonder if a version increment of Eigen should be a breaking change... Though not explicitely mentioned in the versioning file, this is not something I would expect from a minor version change...
Or did we implicitly already require Eigen 3.4.0?

**Comment by paulgessinger** [comment]
@benjaminhuth I believe we started implicitly requiring it when we switched to C++20 by default.

File: Plugins/Detray/include/Acts/Plugins/Detray/DetrayConversionHelper.hpp
Comment (paulgessinger) [inline Plugins/Detray/include/Acts/Plugins/Detray/DetrayConversionHelper.hpp:1]: This file was fully duplicated from the JSON plugin. Any context on this @asalzburger?

File: Plugins/GeoModel/src/detail/GeoUnionDoubleTrdConverter.cpp [hunk 1]
```diff
@@ -47,12 +47,12 @@ bool trapezoidsAreMergeable(const std::vector<Acts::Vector3> &vtxsa,
 namespace Acts::detail {
 
 Result<GeoModelSensitiveSurface> GeoUnionDoubleTrdConverter::operator()(
-    const GeoFullPhysVol &geoFPV, const GeoShapeUnion &geoUnion,
+    const GeoFullPhysVol &geoFPV, const GeoShapeUnion &geoTube,
     const Transform3 &absTransform, bool sensitive) const {
-  const auto shiftA = dynamic_cast<const GeoShapeShift *>(geoUnion.getOpA());
-  const auto shiftB = dynamic_cast<const GeoShapeShift *>(geoUnion.getOpB());
+  const auto shiftA = dynamic_cast<const GeoShapeShift *>(geoTube.getOpA());
```
[the remark this answers is not in the document]
Comment (paulgessinger) [reply Plugins/GeoModel/src/detail/GeoUnionDoubleTrdConverter.cpp]: done

**Comment by paulgessinger** [comment]
Removed the traccc patch in favor of https://github.com/acts-project/traccc/releases/tag/v0.15.0
This should be good to go now @benjaminhuth.

**Review by benjaminhuth — APPROVED** [review]
LGTM!

**Comment by acts-project-service** [comment]
## :white_check_mark: Athena integration test results [df0ab36396d6f03d23ae5beb16c5265dc9b20e82]
### :white_check_mark: All tests successful

|status|job|report|
|------|---|------|
|:green_circle: | [report_pull_request](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/42403210)||
|:green_circle: | [run_art_test: test_data18_13TeV_1000evt](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/42403209)||
|:green_circle: | [run_art_test: test_ttbarPU40_reco](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/42403208)||
|:green_circle: | [test_ActsDumpGeometryIdentifiers](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/42403207)||
|:green_circle: | [test_ActsCheckObjectCountsCachedWorkflow](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/42403206)||
|:green_circle: | [test_ActsCheckObjectCountsWorkflow](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/42403205)||
|:green_circle: | [test_ActsEFTrackFit](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/42403204)||
|:green_circle: | [test_ActsPersistifySeeds](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/42403203)||
|:green_circle: | [test_ActsBenchmarkWithSpot](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/42403202)||
|:green_circle: | [test_ActsAnalogueClustering](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/42403201)||
|:green_circle: | [test_ActsWorkflowHeavyIons](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/42403200)||
|:green_circle: | [test_ActsWorkflowFastTracking](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/42403199)||
|:green_circle: | [test_ActsWorkflowCached](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/42403198)||
|:green_circle: | [test_ActsWorkflow](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/42403197)||
|:green_circle: | [test_ActsValidateAmbiguityResolution](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/42403196)||
|:green_circle: | [test_ActsValidateResolvedTracks](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/42403195)||
|:green_circle: | [test_ActsValidateTracks](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/42403194)||
|:green_circle: | [test_ActsValidateActsCoreSpacePoints](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/42403193)||
|:green_circle: | [test_ActsValidateActsSpacePoints](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/42403192)||
|:green_circle: | [test_ActsValidateSeeds](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/42403191)||
|:green_circle: | [test_ActsValidateOrthogonalSeeds](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/42403190)||
|:green_circle: | [test_ActsValidateClusters](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/42403189)||
|:green_circle: | [test_ActsPersistifyEDM](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/42403188)||
|:green_circle: | [test_ActsGSFRefitting](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/42403187)||
|:green_circle: | [test_ActsKfRefitting](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/42403186)||
|:green_circle: | [test_ActsExtrapolationAlgTest](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/42403185)||
|:green_circle: | [test_ActsITkTest](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/42403184)||
|:green_circle: | [run_workflow_tests_run4_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/42403183)||
|:green_circle: | [run_workflow_tests_run2_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/42403182)||
|:green_circle: | [run_workflow_tests_run2_data](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/42403181)||
|:green_circle: | [run_workflow_tests_run3_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/42403180)||
|:green_circle: | [run_workflow_tests_run3_data](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/42403179)||
|:green_circle: | [run_unit_tests](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/42403178)||

**Pull request merged**